3/3scale-istio-adapter/Dockerfiles/build.sh

```diff
@@ -0,0 +1,75 @@
+# ----------------------------------------------------------------------------
+# Package       : 3scale-istio-adapter 
+# Version       : code-cleanups
+# Source repo   : https://github.com/3scale/3scale-istio-adapter
+# Tested on     : RHEL_8.2
+# Script License: Apache License, Version 2 or later
+# Maintainer    : Krishna Harsha Voora
+#
+# Disclaimer: This script has been tested in non-root mode on given platform using
+#             the mentioned version of the package. It may not work as expected 
+#             with newer versions of the package and/or distribution.
+#             In such case, please contact "Maintainer" of this script.
+# ----------------------------------------------------------------------------
+
+#!/bin/bash
+
+if [ -z $1 ] || [ "$1" == "laststablerelease" ]
+then
+	RELEASE_TAG=code-cleanups
+else
+	RELEASE_TAG=$1
+fi
+
+echo "RELEASE_TAG = $RELEASE_TAG"
+
+# Download/Setup Golang 1.11
+cd /usr/local/
+wget https://golang.org/dl/go1.11.13.linux-ppc64le.tar.gz
+tar -zxf go1.11.13.linux-ppc64le.tar.gz
+rm -f go1.11.13.linux-ppc64le.tar.gz
+
+
+# Update PATH Variables
+export GOPATH=$HOME/ibm/
+export PATH=$GOPATH/bin/:$PATH:/usr/local/go/bin/
+
+# Repository Path
+mkdir -p $GOPATH/bin/
+mkdir -p $GOPATH/src/
+mkdir -p $GOPATH/pkg/
+mkdir -p $GOPATH/src/github.com/3scale/
+cd $GOPATH/src/github.com/3scale/
+
+
+# Install & Setup dep
+curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+
+# Clone Repository
+git clone https://github.com/3scale/3scale-istio-adapter -b $RELEASE_TAG
+cd 3scale-istio-adapter
+
+# Checkout v2.0.2 branch
+# git checkout tags/v2.0.2
+
+# Run dep
+dep ensure -v
+
+# Make
+make build-adapter build-cli
+
+# run tests if "runtest" only if passed as argument
+
+if [ "$2" == "runtest" ]
+then
+
+	# Unit Tests
+	make unit
+
+	# Integration Tests
+	make integration
+fi
+
+# Copy Binaries
+cp _output/3scale-istio-adapter /ws/
+cp _output/3scale-config-gen /ws/
```

Comment:
Maybe a good idea to cleanup after copying binaries. Other than that LGTM.
# cleanup
cd ..
rm -rf 3scale-istio-adapter

Reply:
> Maybe a good idea to cleanup after copying binaries. Other than that LGTM.

I think we may not need this, as we are deleting the container after copying the binaries, so I have not included the step, I can in case it is required.

```docker run --rm -v `pwd`:/ws 3scale-istio-adapter:ibm bash -l -c "cd /ws; ./build.sh <rel_tag> <runtest> | tee output.log"```

Reply:
container is cleaned up but the cloned repo dir would persist on the host at `pwd`/3scale-istio-adapter.
Up to you but maybe we can leave it in this case as repo may not be huge.
Otherwise LGTM.

Reply:
I'm not quite sure where this ended up, but we should always clean up anything we don't need in the container at the end, same thing with the build script.  Maybe don't delete if the build fails, so someone can easily debug without having to reinstall dependencies, but if it succeeds, clean up all but the necessary binaries or equivalent.

Reply:
The requested changes are now in place.